.github/workflows/client.yml

```diff
@@ -0,0 +1,42 @@
+name: Go
+
+#TODO: extend the conditions once workflow gets tested together with other workflows 
+on:  
+  push:
+    branches:
+      - master
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: satackey/action-docker-layer-caching@v0.0.11
+        continue-on-error: true # ignore the failure of a step and avoid terminating the job
+      - name: Run Docker build
+        run: |
+          docker build \
+            --target gobuild \
+            --tag go-build-env .
+          docker build \
+            --tag keep-client . 
+      - name: Create test results directory
+        run: |
+          mkdir -p $GITHUB_WORKSPACE/test-results/
```

Comment:
Maybe we could simplify it and do just `mkdir test-results`? In the next step, we've got to use `$GITHUB_WORKSPACE` to define the volume as a path, but I believe we don't need it here.